src/renderers/dom/shared/devtools/ReactDOMDuplicateSelectValuesDevtool.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactDOMDuplicateSelectValuesDevtool
+ */
+
+'use strict';
+
+var ReactComponentTreeDevtool = require('ReactComponentTreeDevtool');
+var warning = require('warning');
+
+var didWarnDupeSelectValues = false;
+
+function handleElement(debugID, element) {
+  
+  if (didWarnDupeSelectValues) {
+    return;
+  }
+  
+  if (element == null) {
+    return;
+  }
+
+  if (element.type !== 'select') {
+    return;
+  }
+
+  if ((element.props == null) || (element.props.children == null)) {
+    return;
+  }
+
+  // Uncontrolled select elements can have duplicate values.
+  if (element.props.hasOwnProperty('defaultValue')) {
+    return;
+  }
+
+  let values = {};
+  let options = element.props.children;
+
+  //  If options is not iterable make it an array.
+  if (typeof options[Symbol.iterator] !== 'function') {
+    options = [options];
+  }
+
+  for (const option of options) {
+    if (option.type === 'optGroup') {
```

Comment:
It can be any primitive value, especially null/undefined, additionally whitespace strings will not cause warnings in next React, but that's kind of besides the point. You need to explicitly check that it's an element before working on it or the code will throw in some cases.